<launch>
  <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find TurtleKinect)/kinect.vcg" />

  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />

  <!-- <node pkg="nodelet" type="nodelet" name="passthrough_floor" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/camera_top/rgb/points" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: -5 
      filter_limit_max: 2.95
      filter_limit_negative: False
    </rosparam>
  </node> -->


  <node pkg="TurtleKinect" type="floor_segment" name="floor_segment" output="screen">
    <remap from="~input" to="/camera_top/rgb/points" />
    <rosparam>
      ModelType: 0
      MaxIterations: 10
      DistanceThreshold: 0.3
      RatioLimit: 0.8
      ExtractNegative: True
    </rosparam>
  </node>

  <node pkg="nodelet" type="nodelet" name="passthrough_wall_bottom" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="floor_segment/output" />
    <rosparam>
      filter_field_name: y
      filter_limit_min: -0.8
      filter_limit_max: 5.0
      filter_limit_negative: False
    </rosparam>
  </node>


  <node pkg="nodelet" type="nodelet" name="passthrough_wall_top" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/passthrough_wall_bottom/output" />
    <rosparam>
      filter_field_name: y
      filter_limit_min: -5.0
      filter_limit_max: 1.3
      filter_limit_negative: False
    </rosparam>
  </node>

  <node pkg="nodelet" type="nodelet" name="passthrough_wall_left" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/passthrough_wall_top/output" />
    <rosparam>
      filter_field_name: x
      filter_limit_min: -5
      filter_limit_max: 1.5
      filter_limit_negative: False
    </rosparam>
  </node>

  <node pkg="nodelet" type="nodelet" name="passthrough_wall_right" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/passthrough_wall_left/output" />
    <rosparam>
      filter_field_name: x
      filter_limit_min: -1.39
      filter_limit_max: 5.0
      filter_limit_negative: False
    </rosparam>
  </node>

  <node pkg="nodelet" type="nodelet" name="voxel_grid" args="load pcl/VoxelGrid pcl_manager" output="screen">
    <remap from="~input" to="/passthrough_wall_right/output" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: -5
      filter_limit_max: 5.0
      filter_limit_negative: False
      leaf_size: 0.03
    </rosparam>
  </node>

  <node pkg="TurtleKinect" type="turtle_cluster" name="turtle_cluster" output="screen">
    <remap from="~/input" to="/voxel_grid/output" />
    <rosparam>
      ClusterTolerance: 0.08      
      MinClusterSize: 50
      MaxClusterSize: 250000 
    </rosparam>
  </node>

  <node pkg="TurtleKinect" type="ShowCentroidsOnTf.py" name="ShowCentroidsOnTf" output="screen" />
  
</launch>